didman/didman.go

```diff
@@ -67,19 +67,47 @@ func (d *didman) Name() string {
 	return ModuleName
 }
 
-func (d *didman) AddEndpoint(id did.DID, serviceType string, u url.URL) error {
+func (d *didman) AddEndpoint(id did.DID, serviceType string, u url.URL) (*did.Service, error) {
 	logging.Log().Debugf("Adding endpoint (did: %s, type: %s, url: %s)", id.String(), serviceType, u.String())
-	err := d.addService(id, serviceType, u.String(), nil)
+	service, err := d.addService(id, serviceType, u.String(), nil)
 	if err == nil {
 		logging.Log().Infof("Endpoint added (did: %s, type: %s, url: %s)", id.String(), serviceType, u.String())
 	}
-	return err
+	return service, err
+}
+
+func (d *didman) DeleteEndpoint(id did.DID, serviceType string) error {
+	doc, _, err := d.docResolver.Resolve(id, nil)
+	if err != nil {
+		return err
+	}
+
+	var serviceID *ssi.URI
+	for _, s := range doc.Service {
+		if s.Type == serviceType {
+			serviceID = &s.ID
+			break
```

Comment:
duplicate services possible when conflicted

Reply:
Yes, so, delete all of them?